tensorflow_datasets/object_detection/deeplesion.py

```diff
@@ -0,0 +1,257 @@
+            'zipfile53':'https://nihcc.box.com/shared/static/7x4pvrdu0lhazj83sdee7nr0zj0s1t0v.zip',
+            'zipfile54':'https://nihcc.box.com/shared/static/z7s2zzdtxe696rlo16cqf5pxahpl8dup.zip',
+            'zipfile55':'https://nihcc.box.com/shared/static/shr998yp51gf2y5jj7jqxz2ht8lcbril.zip',
+            'zipfile56':'https://nihcc.box.com/shared/static/kqg4peb9j53ljhrxe3l3zrj4ac6xogif.zip',
+            'ann_file':'https://raw.githubusercontent.com/anir16293/Deep-Lesion/master/DL_info.csv'
+    }
+    paths = dl_manager.download_and_extract(resources)
+    ann_path = paths['ann_file']
+    del paths['ann_file']
+    lut = _build_lut(paths)
+    train_split, val_split, test_split = _ann_parser(ann_path)
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            gen_kwargs={
+                "lut": lut,
+                "split": train_split,
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            gen_kwargs={
+                "lut": lut,
+                "split": val_split,
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            gen_kwargs={
+                "lut": lut,
+                "split": test_split,
+            },
+        ),
+    ]
+
+
+  def _generate_examples(self, lut, split):
+    """Returns examples
+    :type lut: dict, lookup dictionary to lookup whole paths of series_folder
+    :type split: pd.dataframe, annotation information associated to images
+    :rtype: yield idx, example
+    """
+    for idx, value in enumerate(split.values):
+      file_name, bboxs, size, sp = value
+      record = {
+          "image/name": file_name,
+          "image": _lookup_image_path(lut, file_name),
+          "bbox": _format_bboxs(bboxs, size)
+      }
+      yield idx, record
+
+    
+def _build_lut(paths):
+    """Returns lookup dictionary to lookup whole paths of series_folder
+    :type paths: dict, {<zipfn>:<unzip_path>}
+    :rtype: dict, {<series_folder>:<unzip_path>/Images_png/<series_folder>}
+    """
+    lut = {}
+    for k, v in paths.items():
+        for fd in tf.io.gfile.listdir(os.path.join(v,'Images_png')):
+            lut[fd] = os.path.join(v,'Images_png', fd)
+    return lut
+        
+
+def _lookup_image_path(lut, file_name):
+    """Returns whole path of an image
+    :type lut: dict, lookup dictionary
+    :file_name: string, image name recorded in the annotation file
+    """
+    slice_fn = file_name.split('_')[-1]
+    fd = '_'.join(file_name.split('_')[:-1])
+    
+    return os.path.join(lut[fd], slice_fn)
+    
+    
+def _format_bboxs(bboxs, size):
+    """Return bbox feature
+    :type bboxs: string, "xxx,xxx,xxx,xxx"
+    :type size: string, "xxx,xxx"
+    :rtype: tfds.features.BBox
+    """
+    size = [float(x) for x in size.split(',')]
+    coords = np.clip([float(x) for x in bboxs.split(',')], 0.0, size[0])
```

Comment:
please comment that you assume that sizes are equal. Or preferably do:
if len(size) != 2 or size[0] != size[1]: 
  raise AssertionError(...)

Reply:
> please comment that you assume that sizes are equal. Or preferably do:
> if len(size) != 2 or size[0] != size[1]:
> raise AssertionError(...)

fixed